lib/spack/spack/test/cmd/common/arguments.py

```diff
@@ -20,6 +20,7 @@ def parser():
     arguments.add_common_arguments(p, ['jobs'])
     yield p
     # Cleanup the command line scope if it was set during tests
+    spack.config.config._get_config_memoized.cache.clear()
```

Comment:
I think any test which sets or changes a scope should be forced to use a `spack.config` module function (which should clear memoizations). I think for example this could just call `spack.config.config.clear_caches()`

Reply:
Done.